### PR TITLE
fix: use date number instead of week number

### PR DESCRIPTION
**Pull request by afifvdin**
## Summary

This PR updates the streak metadata logic to display the actual day of the month (e.g., 15) in the reading progress widget instead of the day index within the week (e.g., 1-7). This ensures users verify their progress against specific dates. Additionally, it removes unused date utility functions (`compareDateWithToday`) and their associated tests.

## Type of Change

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as
      expected)
- [ ] 📝 Documentation update
- [x] ♻️ Refactoring (no functional changes)

## Scope Confirmation

- [x] This PR addresses **one** feature/fix only
- [ ] If multiple changes were needed, they are split into separate PRs

## Rollback Safety

- [x] Can be safely reverted without data issues or migrations
- [ ] Rollback requires special steps (describe below):

## Test Plan

- [x] Unit tests added/updated (Removed obsolete tests)
- [ ] Integration tests added/updated
- [x] Manual testing performed

**Testing steps:**

1. Navigate to the Quran Growth Journey section on the home page.
2. Observe the "Current Week Progress" / streak widget.
3. Verify that the circles display the actual date of the month (e.g., 15, 16) rather than the week index (1, 2...).
4. Confirm that the current day is correctly highlighted.

### Edge Cases Verified

- [x] ⏳ Loading state handled
- [ ] ❌ Error state handled
- [ ] 📭 Empty state handled
- [x] 👤 Logged-in vs guest behavior (if applicable)

### Code Quality

- [x] I have performed a **self-review** of my code (file by file)
- [x] My code follows the [project style guidelines](/.github/copilot-instructions.md)
- [x] No `any` types used (or justified if unavoidable)
- [x] No unused code, imports, or dead code included
- [x] Complex logic has inline comments explaining "why"
- [x] Functions are under 30 lines and follow single responsibility

### Testing & Validation

- [x] All tests pass locally (`yarn test`)
- [x] Linting passes (`yarn lint`)
- [x] Build succeeds (`yarn build`)
- [x] Edge cases and error scenarios are handled

### Documentation

- [x] Code is self-documenting with clear naming
- [ ] README updated (if adding features or setup changes)
- [ ] Inline comments added for complex logic

### Localization (if UI changes)

- [x] All user-facing text uses `next-translate`
- [x] Only English locale files modified (Lokalise handles others)
- [x] RTL layout verified

### Accessibility (if UI changes)

- [x] Semantic HTML elements used
- [x] ARIA attributes added where needed
- [x] Keyboard navigation works

## Screenshots/Videos

| Before | After |
| ------ | ----- |
| *(Shows 1, 2, 3...)* | *(Shows e.g. 15, 16, 17...)* |

## Reviewer Notes

Removed `compareDateWithToday` and related tests in [src/utils/datetime.ts](cci:7://file:///home/afifvdin/Desktop/quran.com-frontend-next/src/utils/datetime.ts:0:0-0:0) as they were no longer used.

## AI Assistance Disclosure

- [x] AI tools were NOT used for this PR
- [ ] AI tools were used, and I have **thoroughly reviewed and validated** all generated code

### CODE REVIEW

File: src/components/HomePage/QuranGrowthJourneySection/CollapsibleSection/QuranReadingGoals/CurrentWeekProgress.tsx [hunk 1]
```diff
@@ -19,29 +19,27 @@ const CurrentWeekProgress: React.FC<Props> = ({ weekData, goal }) => {
...
 
     // if the user has a goal, we want to show a checked circle if the user has completed his goal for the day
     // otherwise, we want to show a checked circle if the user has read at all for the day
     const isGoalDone = goal ? convertFractionToPercent(readingDay?.progress || 0) >= 100 : hasRead;
 
-    if (isGoalDone) return [DayState.Checked, isToday];
+    if (isGoalDone) return DayState.Checked;
 
-    if (normalizedDate > today) return [DayState.Future, isToday];
+    if (day.date > new Date()) return DayState.Future;
```
[the remark this answers is not in the document]
Comment (afifvdin) [reply src/components/HomePage/QuranGrowthJourneySection/CollapsibleSection/QuranReadingGoals/CurrentWeekProgress.tsx:32]: yes

[the remark this answers is not in the document]
Comment (afifvdin) [reply src/components/HomePage/QuranGrowthJourneySection/CollapsibleSection/QuranReadingGoals/CurrentWeekProgress.tsx:32]: yes
